test/core/test_tensor_wrapper.py

```diff
@@ -15,15 +17,24 @@ def test_smoke(self, device, dtype):
         assert tensor.shape == (1, 2, 3, 4)
         assert tensor.device == device
         assert tensor.dtype == dtype
-        self.assert_close(data, tensor.data)
-        self.assert_close(data, unwrap(tensor))
+        self.assert_close(data, tensor.unwrap())
+
+    def test_serialization(self, device, dtype):
+        data = torch.rand(1, 2, 3, 4, device=device, dtype=dtype)
+        tensor: TensorWrapper = wrap(data, TensorWrapper)
+
+        with tempfile.NamedTemporaryFile() as tmp:
+            file_path = tmp.name + ".pt"
+            torch.save(tensor, file_path)
+            loaded_tensor: TensorWrapper = torch.load(file_path)
+            self.assert_close(loaded_tensor.unwrap(), tensor.unwrap())
 
     def test_wrap_list(self, device, dtype):
         data_list = [torch.rand(2, device=device, dtype=dtype), torch.rand(3, device=device, dtype=dtype)]
         tensor_list = wrap(data_list, TensorWrapper)
         assert isinstance(tensor_list, list)
         assert len(tensor_list) == 2
-        tensor_list_data = unwrap(tensor_list)
+        tensor_list_data = tensor_list.unwrap()
```

Comment:
not sure of the idea here

Reply:
It will return the data structure as plane tensor. For now is a requirement of assert allclose, and some functions I detected that torch doesn’t like wrappers of tensors

Reply:
But should this be a list of tensors, or a tensor itself? -- asking because this was asserted to be a list, and a list will not have this method

Reply:
oh you are right. I think we can decide the behavior here but the general idea for the wrapper is that we cast Any to have a Tensor behavior. Open to ideas here

Reply:
Well, just if we try to lead internally with having a sequence of tensor which I think probably will be tricky

otherwise, we can do like is with tensors, something like:
```python
data_list = [
    5, 6.0, [7.0], # i think this should work too, no?
   torch.rand(2, device=device, dtype=dtype), torch.rand(3, device=device, dtype=dtype)
]

a = wrap(data_list, TensorWrapper)
assert isinstance(a, Sequence) and all(isinstance(x, TensorWrapper) for x in a)

b = []
for i in range(len(a)):
    # do something to each tensor
    b.append(a[i].unwrap())

assert  isinstance(b, list) and all(isinstance(x, Tensor) for x in a)
```

I think this second assert will be true if each item be a `TensorWrapper` too, because inheritance